File: tutorials/cpp/tutorial_oxidation.cpp
#include "../../src/cpp/core/wafer.hpp"
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"
#include "../../src/cpp/modules/oxidation/oxidation_model.hpp"
#include <memory>
#include <iostream>

int main() {
  std::cout << "Tutorial: Oxidation Module\n";
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  std::cout << "Step 1: Initialize grid\n";
  geometry.initializeGrid(wafer, 100, 100);
  OxidationModel oxidation;
  std::cout << "Step 2: Run thermal oxidation\n";
  oxidation.simulateOxidation(wafer, 1000.0, 3600.0);
  std::cout << "Final thickness: " << wafer->getThickness() << " um\n";
  return 0;
}

File: tutorials/python/tutorial_oxidation.py
from semiconductor_simulator.simulator import Simulator

print("Tutorial: Oxidation Module")
simulator = Simulator()
print("Step 1: Initialize grid")
simulator.initialize_geometry(100, 100)
print("Step 2: Run thermal oxidation")
simulator.run_oxidation(1000.0, 3600.0)
print("Step 3: Visualize oxide layer")
simulator.initialize_renderer(400, 400)
simulator.render(simulator.get_wafer())